ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesfragment/services/RootFragmentCreatorImpl.java

```diff
@@ -18,16 +17,13 @@ public RootFragmentCreatorImpl(LdesFragmentRepository ldesFragmentRepository) {
 
 	@Override
 	public LdesFragment createRootFragmentForView(String viewName) {
-		Optional<LdesFragment> optionalRoot = ldesFragmentRepository
-				.retrieveRootFragment(viewName);
-		if (optionalRoot.isEmpty()) {
-			return createRoot(viewName);
-		}
-		return optionalRoot.get();
+		return ldesFragmentRepository
+				.retrieveRootFragment(viewName)
+				.orElseGet(() -> createRoot(viewName));
 	}
 
 	private LdesFragment createRoot(String viewName) {
-		LdesFragment ldesFragment = new LdesFragment("collectionName", viewName, List.of());
+		LdesFragment ldesFragment = new LdesFragment(viewName.split("/")[0], viewName, List.of());
```

Comment:
Could be an idea to create a value object ViewName that can be retrieved from ViewSpecification and that has two methods:
* getEntireViewName(){return viewName;}
* getCollectionName(){return viewName.split("/")[0];}

Reply:
I agree using a ViewName class would give us a lot of value all over the application. This is a bit more work so I will add it in a following PR https://vlaamseoverheid.atlassian.net/jira/software/projects/VSDSPUB/boards/681?selectedIssue=VSDSPUB-620